cmd/download_helper.go

```diff
@@ -0,0 +1,244 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	netURL "net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/exercism/cli/api"
+	"github.com/exercism/cli/config"
+	"github.com/exercism/cli/workspace"
+	"github.com/spf13/viper"
+)
+
+type downloadParams struct {
+	usrCfg *viper.Viper
+	uuid   string
+	slug   string
+	track  string
+	team   string
+}
+
+type downloadPayloadContext struct {
```

Comment:
I did this here https://github.com/exercism/cli/pull/756/commits/02cbe9bf73f417ad7f53f69c4e944c68b3a3e659. Unless I'm missing something, I don't think there is an additional benefit of merging `downloadParams` into `downloadContext`. The decision rests on if you think that having a separate parameter object is worthwhile. I think it might be worth having for separation of concerns. What do you think?

Reply:
Yepp, I like it!